llvm/unittests/Support/ProgramTest.cpp

```diff
@@ -583,10 +583,14 @@ TEST_F(ProgramEnvTest, TestLockFileExclusive) {
     int FD2;
     ASSERT_NO_ERROR(fs::openFileForReadWrite(LockedFile, FD2,
                                              fs::CD_OpenExisting, fs::OF_None));
+    // File should be non-exclusive locked. Try acquire exclusive lock will fail
+    // but non-exclusive will succeed.
+    EXPECT_TRUE(
+        fs::tryLockFile(FD2, std::chrono::seconds(0), /*Exclusive=*/true));
+
+    EXPECT_FALSE(
+        fs::tryLockFile(FD2, std::chrono::seconds(0), /*Exclusive=*/false));
```

Comment:
The logic here looks backwards from the comments above. Seems to me like you take the exclusive lock first, then try to take the non-exclusive lock after.

Reply:
I think the comment is correct. Exclusive fail first and non-exclusive succeed second. I think the confusion is the first sentence, which should actually states `file is currently non-exclusively locked by other processes`.

Reply:
OMG. It returns an Error. 🤦   I've inverted the logic when I made the comment, thinking the exclusive lock succeeded. Sorry for the noise. You're right that this seems correct.